dev-docs/RFCs/v4.0.0/terms_for_concepts_and_datatypes.md

```diff
@@ -0,0 +1,60 @@
+# RFC: Terms for H3 Concepts and Data Types
+
+- **Authors**: AJ Friend
+- **Date**: 2020-02-02
+- **Status**: Work In Progress
+
+## Motivation
+
+Concepts like `hexagon`, `cell`, and `index` are currently used ambiguously
+in the H3 codebase and in documentation. This can cause confusion, for example,
+when a function might only work on strict H3 *hexagons*, but fails when
+encountering H3 *pentagons*.
+
+Reaching a consensus on the precise, technical language used when discussing H3 concepts will clarify future library discussions and improve end-user documentation.
+
+## Proposals
+
+## Concepts
+
+The following technical terms should be used in the documentation, the H3 codebase, and precise technical discussions of the library.
+
+- `H3Index`, "H3 index", or "index"
+    - an unsigned 64-bit integer representing **any** H3 object (hexagon, pengaton, unidirectional edge, ...)
+    - often represented as a 15-character (or 16-character) hexadecimal string, like `'8928308280fffff'`
+- cell
```

Comment:
Nit: Should `H3 cell` be a preferred form? One question here and above is when to use `H3 index|cell` and when to use the short form, particularly in function names.

Reply:
I've added some new notes to the terminology section. Let me know if you think that covers it. I'll resolve this for now.